solr/solrj/src/java/org/apache/solr/common/cloud/ZkClientConnectionStrategy.java

```diff
@@ -121,4 +122,35 @@ protected SolrZooKeeper createSolrZooKeeper(
 
     return result;
   }
+
+  // Override for testing
+  protected ZooKeeper newZooKeeperInstance(
+      final String serverAddress, final int zkClientTimeout, final Watcher watcher)
+      throws IOException {
+    return new ZooKeeper(serverAddress, zkClientTimeout, watcher);
+  }
+
+  /**
+   * Instantiate a new connection strategy for the given class name
+   *
+   * @param name the name of the strategy to use
+   * @return the strategy instance, or null if it could not be loaded
+   */
+  public static ZkClientConnectionStrategy forName(String name, ZkClientConnectionStrategy def) {
+    log.debug("Attempting to load zk connection strategy '{}'", name);
+    if (name == null) {
+      return def;
+    }
+
+    try {
+      // TODO should this use SolrResourceLoader?
```

Comment:
Looks like `ZkController` via `CoreContainer` would have access to `SolrResourceLoader` but `SolrZkClient` wouldn't have since `SolrResourceLoader` isn't solrj.